packages/grpc-js/src/server.ts

```diff
@@ -870,15 +900,15 @@ export class Server {
     this.trace('Closing session initiated by ' + session.socket?.remoteAddress);
     const sessionInfo = this.sessions.get(session);
     const closeCallback = () => {
-      if (this.channelzEnabled && sessionInfo) {
+      if (sessionInfo) {
         this.sessionChildrenTracker.unrefChild(sessionInfo.ref);
         unregisterChannelzRef(sessionInfo.ref);
+        this.sessions.delete(session);
```

Comment:
I believe this is redundant with the same operation at the end of `_channelzSessionHandler`.